feat(invity): Dex exchange crypto functionality

martinboehm opened the pull request:
This PR adds a possibility to exchange ETH and ERC20 tokens in a decentralized exchange using smart contract provided by 1Inch.

The functionality extends the existing Exchange Crypto dialogs in the Trade section of the account. If the selected account is ETH (user can choose ETH or ERC20 token as a source) and the target coin is ETH or ERC20, the system gets a DEX quote in addition to the traditional floating/fixed rates quotes.
![image](https://user-images.githubusercontent.com/11609674/142742637-4e658bbe-f943-44e0-a6c5-52c1a6754260.png)

When the DEX quote is selected, the user proceeds through the target account selection (like in standard exchange) and then to two new steps:  
1) approval transaction step (necessary to approve 1Inch's contract to access the user's tokens)
![image](https://user-images.githubusercontent.com/11609674/142742616-343dc153-d238-4af2-b9e1-8d1d4cb00bb6.png)

2) the swap transaction step
![image](https://user-images.githubusercontent.com/11609674/142742659-3338e466-9e7a-4270-a05a-a13bf0d58a50.png)

tomasklim commented on patches/@types+invity-api+1.0.3.patch:
```diff
@@ -0,0 +1,86 @@
+diff --git a/node_modules/@types/invity-api/index.d.ts b/node_modules/@types/invity-api/index.d.ts
```
Maybe release `1.0.5` types?

martinboehm replied:
We will certainly release them, however it is not certain that they manage to approve it fast enough. When it is released, we will remove the patch.

martinboehm replied:
@tomasklim Regarding the design: while your input is certainly welcome and I will forward it in our next discussion with the designer (Ondra), we cannot modify it based on the input from a single reviewer. We would not get anywhere as everyone has a different aesthetic sense. For example the Receiving address is not even part of this PR.

martinboehm replied:
Upgraded invity-api to 1.0.5, DefinitelyTyped worked quite fast! Also modified the padding in the fees section.

tomasklim commented on packages/suite/src/support/messages.ts:
```diff
@@ -449,6 +491,114 @@ export default defineMessages({
+    },
+    TR_EXCHANGE_SWAP_SLIPPAGE_INFO: {
+        defaultMessage:
+            'Slippage is the difference between the expected price of a trade and the price at which the trade is executed',
```
As a user, I am not sure if I understand the info correctly. I think it would be better to describe it on amount.
e.g. 
`Maximum acceptable % difference between the expected amount of token and the amount you actually receive...` 

I would also add there minimum received amount
![Screenshot 2021-11-22 at 12 39 13](https://user-images.githubusercontent.com/33235762/142855405-54746c90-bf1b-4069-9c9d-dafa72b85f5a.png)

martinboehm replied:
@tomasklim That is a good idea, we will think how to display the minimum received amount.

martinboehm replied:
@tomasklim We updated the tooltip text and added a Slippage summary part, thank you for your tip!

tomasklim reviewed (changes requested):
Fee info looks unaligned
![Screenshot 2021-11-22 at 11 45 50](https://user-images.githubusercontent.com/33235762/142859036-542d9164-add6-4b1d-a613-b81b9ed9d6cb.png)

Receiving address input is small
![Screenshot 2021-11-22 at 11 46 44](https://user-images.githubusercontent.com/33235762/142859123-4358764f-ba15-4649-93d8-cfebc751e727.png)

tomasklim reviewed (approved):
Great feature!

_Tip:_ As 1inch is an aggregator of `Uniswap`, `Sushiswap`,... which have a lot of tokens. You should think about an option to add a custom ERC-20 token (`0x...) to the Exchange tab.

martinboehm commented:
@tomasklim  Thank you! Squashed and rebased over develop. 

On arbitrary contract: I think 1inch actually does not accept just any contract, they provide us a list of supported contracts (which we shorten at this moment to the contracts we support). However, the list of supported coins (in the receive select) is controlled by our server, we can quite easily extend it without the need to wait for Suite release. We will observe the demand and extend it if necessary.
